evalml/tests/pipeline_tests/test_component_graph.py

```diff
@@ -2737,7 +2744,7 @@ def test_get_component_input_logical_types():
         "cat": Categorical(),
         "numeric": Integer(),
         "EMAIL_ADDRESS_TO_DOMAIN(email)": Categorical(),
-        "IS_FREE_EMAIL_DOMAIN(email)": Categorical(),
+        "IS_FREE_EMAIL_DOMAIN(email)": BooleanNullable(),
```

Comment:
The change to not automatically convert all features to `Categorical` will result in `BooleanNullable` columns being introduced into `X` via the `EmailFeaturizer` after the `ReplaceNullableTypes` transformer has been used. 

Because of that, this change will have to wait to be introduced with the rest of the nullable type handling. 😢